### PR TITLE
Remove compile warnings

### LINKED ISSUE #255: Clear up compiler warnings
Compiling our contracts currently returns a large number of compiler warnings. 

Fixing them should be a fairly simple task, but will require some time to go through everything. 

**Recommended approach:**

IMO the solc binary give much easier to read colored output compared to hardhat, so install `solc` 
1. Install `solc` verson 0.7.6 following [these instructions](https://docs.soliditylang.org/en/v0.7.6/installing-solidity.html) (any of the methods described on that page _other than_ `npm i -g solc` because `solcjs != solc`!). 
2. `cd` into the root of this repo. 
3. Run the compiler on the full system using: `solc --allow-paths . contracts/optimistic-ethereum/**/*.sol` 
4. Work through the warnings systematically. You might choose to focus on one file at a time, or one type of warning at a time, whichever works for you. 

I would suggest submitting a draft PR for early review after clearing up the first dozen or so warnings, just to be sure we're on track.

### PR DESCRIPTION
**Pull request by vaibhavchellani**
## Description
Addresses compiler warnings.
### Fixes
- Fixes #255 

## Contributing Agreement

- [x] I have read and understood the [Optimism Contributing Guide and Code of Conduct](https://github.com/ethereum-optimism/optimism-monorepo/blob/master/.github/CONTRIBUTING.md) and am following those guidelines in this pull request.

### CODE REVIEW

**Comment by vaibhavchellani** [comment]
Created a draft PR to check with @maurelian if we are on the same page. 

Let me know if it's all good and I will send more commits to remove other warnings.

File: contracts/optimistic-ethereum/libraries/resolver/Lib_AddressResolver.sol [hunk 1]
```diff
@@ -25,7 +25,7 @@ contract Lib_AddressResolver {
      */
     constructor(
         address _libAddressManager
-    ) public {
+    )  {
```
Comment (maurelian) [inline contracts/optimistic-ethereum/libraries/resolver/Lib_AddressResolver.sol:28]: Let's also mark this contract as `abstract`.

File: contracts/optimistic-ethereum/libraries/utils/Lib_ReentrancyGuard.sol [hunk 2]
```diff
@@ -34,7 +34,7 @@ abstract contract Lib_ReentrancyGuard {
 
     uint256 private _status;
 
-    constructor () internal {
+    constructor () {
```
Comment (maurelian) [inline contracts/optimistic-ethereum/libraries/utils/Lib_ReentrancyGuard.sol:37]: Please also mark this contract as `abstract`.

**Review by maurelian — CHANGES_REQUESTED** [review]
Looking good to me, thank you very much for taking this on! 🙏

**Comment by vaibhavchellani** [comment]
@maurelian here are some questions where some help is appreciated:

- what should we rename [push2](https://github.com/ethereum-optimism/contracts/blob/master/contracts/optimistic-ethereum/libraries/utils/Lib_RingBuffer.sol#L121) to?
- how to remove `Unused function parameter` warnings in the mock contracts, or should we let it be? 
- should we also attempt to remove warnings originating from `UniswapV2ERC20`

**Comment by maurelian** [comment]
> * what should we rename [push2](https://github.com/ethereum-optimism/contracts/blob/master/contracts/optimistic-ethereum/libraries/utils/Lib_RingBuffer.sol#L121) to?

Let's try `pushTwo`, and we'll see what the other maintainers think when the full PR is ready for review.

> * how to remove `Unused function parameter` warnings in the mock contracts, or should we let it be?

Good question. The compiler says `Warning: Unused function parameter. Remove or comment out the variable name to silence this warning.`
So I think we should follow that directive. Here's an example of a diff that fixes the warning in `Abs_BaseCrossDomainMessenger.sol`. You also need to modify the natspec comments to remove the param. 

image




> * should we also attempt to remove warnings originating from `UniswapV2ERC20`

Yes, I think so.

**Comment by maurelian** [comment]
> > * what should we rename [push2](https://github.com/ethereum-optimism/contracts/blob/master/contracts/optimistic-ethereum/libraries/utils/Lib_RingBuffer.sol#L121) to?
> 
> Let's try `pushTwo`, and we'll see what the other maintainers think when the full PR is ready for review.

I'm sorry, I've changed my mind about that. Let's not change `push2` for now.

**Comment by vaibhavchellani** [comment]
I have some changes to push, do you prefer me to send those as a 2nd PR?

**Review by maurelian — APPROVED** [review]

**Pull request merged**